public/app/features/templating/template_srv.ts

```diff
@@ -41,10 +41,27 @@ export class TemplateSrv {
     return this.builtIns.__interval.value;
   }
 
+  /**
+   * @deprecated: this instance variable should not be used and will be removed in future releases
+   *
+   * Use getVariables function instead
+   */
+  get variables(): any[] {
+    return this.getVariables();
+  }
```

Comment:
You can also use https://github.com/grafana/grafana/blob/master/packages/grafana-data/src/utils/deprecationWarning.ts if you want to be more verbose in the console. The only problem with this function is it's going to be called for every getter invocation what may flood the console output. But we could tweak it to only log once per warning.

Reply:
Interesting, didn't know about this one. If I read the code correctly, it will only warn every 10s did I understand that correctly?

Reply:
ah yeas,  haven't noticed it was tweaked already, great!